c10/util/StringUtil.cpp

```diff
@@ -0,0 +1,43 @@
+#include "c10/util/StringUtil.h"
+#include "c10/util/Exception.h"
+
+#include <cstring>
+#include <string>
+
+namespace c10 {
+
+namespace detail {
+
+std::string StripBasename(const std::string& full_path) {
+  const char kSeparator = '/';
+  size_t pos = full_path.rfind(kSeparator);
+  if (pos != std::string::npos) {
+    return full_path.substr(pos + 1, std::string::npos);
+  } else {
+    return full_path;
+  }
+}
+
+} // namespace detail
+
+std::ostream& operator<<(std::ostream& out, const SourceLocation& loc) {
+  out << loc.function << " at " << loc.file << ":" << loc.line;
+  return out;
+}
+
+size_t ReplaceAll(std::string& s, const char* from, const char* to) {
+  AT_CHECK(from && *from, "");
+  AT_CHECK(to, "");
+
+  size_t numReplaced = 0;
+  std::string::size_type lenFrom = std::strlen(from);
+  std::string::size_type lenTo = std::strlen(to);
+  for (std::string::size_type pos = s.find(from); pos != std::string::npos;
```

Comment:
nit: loop index variables are a perfect place for increasing readability and reducing error likelihood with `auto`.

Reply:
Good point. will do.